FAILED ert_tests/ensemble_evaluator/test_ensemble_legacy.py::test_run_and_cancel_legacy_ensemble

**Describe the bug**
Failed github action on PR.

https://github.com/equinor/ert/actions/runs/1793572277

`FAILED ert_tests/ensemble_evaluator/test_ensemble_legacy.py::test_run_and_cancel_legacy_ensemble`
```

 =================================== FAILURES ===================================
_____________________ test_run_and_cancel_legacy_ensemble ______________________

tmpdir = local('/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/test_run_and_cancel_legacy_ens0')
make_ensemble_builder = <function make_ensemble_builder.<locals>._make_ensemble_builder at 0x133ab0a60>

    @pytest.mark.timeout(60)
    def test_run_and_cancel_legacy_ensemble(tmpdir, make_ensemble_builder):
        num_reals = 10
        custom_port_range = range(1024, 65535)
        with tmpdir.as_cwd():
            ensemble = make_ensemble_builder(tmpdir, num_reals, 2, job_sleep=5).build()
            config = EvaluatorServerConfig(custom_port_range=custom_port_range)
    
            evaluator = EnsembleEvaluator(ensemble, config, 0, ee_id="1")
    
            with evaluator.run() as mon:
                cancel = True
                for _ in mon.track():
                    if cancel:
                        mon.signal_cancel()
                        cancel = False
    
>           assert evaluator._ensemble.get_status() == state.ENSEMBLE_STATE_CANCELLED
E           AssertionError: assert 'Stopped' == 'Cancelled'
E             - Cancelled
E             + Stopped

ert_tests/ensemble_evaluator/test_ensemble_legacy.py:52: AssertionError
```

Also logged 

    WARNING  ert_shared.ensemble_evaluator.ensemble.base:base.py:53 Illegal state transition from Cancelled to Stopped

Which is a hint that perhaps this test was always flaky, or that cancelling doesn't work properly.

**To Reproduce**
Steps to reproduce the behavior:
1. Push to a PR

**Expected behavior**
Dont fail


**Enviromment**
 - 3.8 macos-latest

--- a/ert_shared/ensemble_evaluator/ensemble/legacy.py
+++ b/ert_shared/ensemble_evaluator/ensemble/legacy.py
@@ -196,8 +196,6 @@ class _LegacyEnsemble(_Ensemble):
         asyncio.set_event_loop(asyncio.new_event_loop())
         logger.debug("cancelling, waiting for wakeup...")
         self._allow_cancel.wait()
-        logger.debug("got wakeup, killing all jobs...")
-        self._job_queue.kill_all_jobs()
         logger.debug("cancelling futures...")
         if self._aggregate_future.cancelled():
             logger.debug("aggregate future was already cancelled")

--- a/res/job_queue/queue.py
+++ b/res/job_queue/queue.py
@@ -458,18 +458,17 @@ class JobQueue(BaseCClass):
                     if not self.is_active() or self.stopped:
                         break
             except asyncio.CancelledError:
-                if self.stopped:
-                    logger.debug(
-                        "observed that the queue had stopped after cancellation, stopping jobs..."
-                    )
-                    self.stop_jobs()
-                    logger.debug("jobs now stopped (after cancellation)")
+                self.kill_all_jobs()  # make available_capacity() return false
+                logger.debug("queue cancelled, stopping jobs...")
+                await self.stop_jobs_async()
+                logger.debug("jobs stopped, re-raising CancelledError")
                 raise
 
             if self.stopped:
                 logger.debug("observed that the queue had stopped, stopping jobs...")
                 await self.stop_jobs_async()
                 logger.debug("jobs now stopped")
+
             self.assert_complete()
             self._differ.transition(self.job_list)
             await JobQueue._publish_changes(ee_id, self._differ.snapshot(), websocket)

--- a/tests/ert_tests/ensemble_evaluator/test_ensemble_legacy.py
+++ b/tests/ert_tests/ensemble_evaluator/test_ensemble_legacy.py
@@ -42,7 +42,7 @@ def test_run_and_cancel_legacy_ensemble(tmpdir, make_ensemble_builder):
     num_reals = 10
     custom_port_range = range(1024, 65535)
     with tmpdir.as_cwd():
-        ensemble = make_ensemble_builder(tmpdir, num_reals, 2, job_sleep=5).build()
+        ensemble = make_ensemble_builder(tmpdir, num_reals, 2, job_sleep=30).build()
         config = EvaluatorServerConfig(
             custom_port_range=custom_port_range, custom_host="127.0.0.1"
         )

